src/StatusBar.ts

```diff
@@ -1,72 +1,254 @@
-import { window, StatusBarAlignment } from 'vscode'
+import * as vscode from 'vscode'
 import * as elegantSpinner from 'elegant-spinner'
+import { extensionName } from './appGlobals'
+import { JestExt } from './JestExt'
+
+export enum StatusType {
+  active,
+  summary,
+}
 
 type Status = 'running' | 'failed' | 'success' | 'stopped' | 'initial'
-type QueueItem = {
+interface StatusUpdateRequest {
   source: string
   status: Status
   details: string | undefined
 }
 
+interface SpinnableStatusBarItem {
+  readonly type: StatusType
+  command: string | undefined
+  text: string | undefined
+  tooltip: string | undefined
+
+  show(): void
+  hide(): void
+  clearSpinner(): void
+  startSpinner(callback: () => void): void
+}
+
+const createStatusBarItem = (type: StatusType, priority: number): SpinnableStatusBarItem => {
+  let spinner: NodeJS.Timer | undefined
+  const item = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Left, priority)
+  const clearSpinner = () => {
+    if (spinner) {
+      clearInterval(spinner)

```

Comment:
in statusBar `render()` , it did call the `clearSpinner()` so basically each `startSpinner` only call back for 1 interval unless it is started again within `render()`. Does that address your conern?

Reply:
My idea was to look at the statusBarItem as a separate object. If for example the implementation of `StatusBar` would some when change and by chance call `startSpinner` twice, the item would spin indefinitely. Therefore I would check within `startSpinner` that `spinner` is `undefined`. But that's just my personal opinion and to be honest completely unimportant.

Reply:
i see what you mean, and I think you are right, it is definitely more defensible for future development, in case spinner did get misused... added `clearSpinner` at the beginning of each `startSpinner` should do the trick.